### PR TITLE
chore(pkg) templatize mirror-scripts on the 'pkgrepo' profile

### PR DESCRIPTION
**Pull request by dduportal**
Blocked by #3358 (and will need rebase before complete review or only check the last commit)

Follow up of https://github.com/jenkins-infra/helpdesk/issues/3414

This PR is the second step of https://github.com/jenkins-infra/jenkins-infra/issues/3352 (after #3358).
It introduces the following changes:

    The set of "mirror scripts" imported in https://github.com/jenkins-infra/jenkins-infra/pull/3358 have their content templatized to map to puppet hieradata configuration used by profiles.
    
Note: the profile `updatesite` has been updated (default param. value move to hieradata) to allow other profiles to lookup the value (`updatesite` and `pkgrepo`) are still highly coupled on the same VM).

Deployment process:

- Disable the puppet agent
- Merge
- Run a noop and check the resulting scripts
- Apply
- Run a `sync.sh` (like the crontab) and validate full execution (~2 min)
- Verify that update-center2 on trusted. does not fail a brand new build (~5 min)
- Report change here

### CODE REVIEW

File: dist/profile/manifests/pkgrepo.pp [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
-  Hash $authorized_ssh_keys             = {},
+  Stdlib::Absolutepath $www_basedir         = '/var/www',
+  String $pkg_basedir                       = 'pkg.jenkins.io',
+  String $pkg_staging_basedir               = 'pkg.jenkins.io.staging',
+  Stdlib::Absolutepath $release_root        = '/srv/releases/jenkins',
+  Stdlib::Fqdn $repo_fqdn                   = 'pkg.origin.jenkins.io',
+  Stdlib::Fqdn $repo_legacy_fqdn            = 'pkg.jenkins-ci.org',
+  Stdlib::Fqdn $mirror_fqdn                 = 'mirrors.jenkins.io',
+  Stdlib::Absolutepath $mirror_home_dir     = '/srv/releases',
+  String $mirror_git_remote                 = 'https://github.com/jenkins-infra/mirror-scripts.git',
```
Comment (smerle33) [inline dist/profile/manifests/pkgrepo.pp]:
```suggestion
```

File: dist/profile/manifests/pkgrepo.pp [hunk 2]
```diff
@@ -51,6 +51,31 @@
...
+    'username' => lookup('archives_jenkins_io_mirroring::username'),
+    'privkey'  => lookup('archives_jenkins_io_mirroring::privkey'),
+    'keypath'  => "${mirror_home_dir}/.ssh/archives",
+  }
+
+  # Used by mirror-scripts
+  $updates_docroot = lookup('profile::updatesite::docroot')
+  $azure_storage_env_file = "${mirror_home_dir}/.azure-storage-env"
+
+  # Used by apache2 and epos
```
Comment (smerle33) [inline dist/profile/manifests/pkgrepo.pp]:
```suggestion
  # Used by apache2 and scripts
```

File: dist/profile/manifests/pkgrepo.pp [hunk 3]
```diff
@@ -86,7 +111,7 @@
     owner   => $mirror_user,
     group   => $mirror_group,
     mode    => '0600',
-    content => lookup('archives_privkey'),
+    content => lookup('archives_jenkins_io_mirroring::privkey'),
```
Comment (smerle33) [inline dist/profile/manifests/pkgrepo.pp]:
```suggestion
    content => $archives_jenkins_io_mirroring['privkey'],
```

File: spec/classes/profile/pkgrepo_spec.rb [hunk 4]
```diff
@@ -36,8 +38,13 @@
     }
   end
 
+  before(:each) do
+    $pkg_docroot = "#{params[:www_basedir]}/#{params[:pkg_basedir]}"
+    # @data = get_data_from_file  # [ '42', '36' ]
```
Comment (smerle33) [inline spec/classes/profile/pkgrepo_spec.rb]:
```suggestion
```

**Review by smerle33 — APPROVED** [review]
LGTM 🚀

**Comment by dduportal** [comment]
Deployed with success during a pairing session with @smerle33 (hence the self-merged suggestions).

We confirmed that the scripts are running properly (as per the described procedure).

**Pull request merged**